cmake/submodules.cmake

```diff
@@ -17,20 +17,33 @@ IF(GIT_EXECUTABLE AND EXISTS "${CMAKE_SOURCE_DIR}/.git")
     ${GIT_EXECUTABLE} config cmake.update-submodules yes")
   ELSEIF(git_config_get_result EQUAL 128)
     SET(update_result 0)
-  ELSEIF (cmake_update_submodules MATCHES force)
-    MESSAGE(STATUS "Updating submodules (forced)")
-    EXECUTE_PROCESS(COMMAND "${GIT_EXECUTABLE}" submodule update --init --force --recursive --depth=1
-                    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
-                    RESULT_VARIABLE update_result)
-  ELSEIF (cmake_update_submodules MATCHES yes)
-    EXECUTE_PROCESS(COMMAND "${GIT_EXECUTABLE}" submodule update --init --recursive --depth=1
-                    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
-                    RESULT_VARIABLE update_result)
   ELSE()
-    MESSAGE(STATUS "Updating submodules")
-    EXECUTE_PROCESS(COMMAND "${GIT_EXECUTABLE}" submodule update --init --recursive --depth=1
+    EXECUTE_PROCESS(COMMAND "${GIT_EXECUTABLE}" --version
                     WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
-                    RESULT_VARIABLE update_result)
+                    OUTPUT_VARIABLE git_version
+                    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    SET(UPDATE_SUBMODULES_COMMAND
+        "${GIT_EXECUTABLE}" submodule update --init --recursive)
+    # Old Git may not work with "--depth 1".
+    # See also: https://github.com/git/git/commit/fb43e31f2b43076e7a30c9cd00d0241cb8cf97eb
+    IF(git_version VERSION_EQUAL "2.8.0" OR git_version VERSION_GREATER "2.8.0")
```

Comment:
This doesn't work:
```
$ git --version 
git version 2.34.1
```
note, that the output starts with a "git version" prefix. Please, test your code both on old and new git to make sure it produces the expected result in all cases.

Also, I'd recommend to use
```cmake
IF (NOT git_version VERSION_LESS "2.8.0")
```
to not repeat the version twice (and because it's shorter)

Reply:
Also, there is GIT_VERSION_STRING , provided to you by FIND_PACKAGE(Git).  the version of git found (since CMake 2.8.8)
Thus  no need to invoke git once more  to find out its version.

Reply:
Oh, sorry. I should have tested with debug print too...

I didn't know `GIT_VERSION_STRING`. Thanks!!!

I've changed to use `GIT_VERSION_STRING` and confirmed this change with a debug print on CI and local:

```diff
diff --git a/cmake/submodules.cmake b/cmake/submodules.cmake
index 34dcfbea77a..8ae96dfb50c 100644
--- a/cmake/submodules.cmake
+++ b/cmake/submodules.cmake
@@ -25,6 +25,7 @@ IF(GIT_EXECUTABLE AND EXISTS "${CMAKE_SOURCE_DIR}/.git")
     IF(NOT GIT_VERSION_STRING VERSION_LESS "2.8.0")
       SET(UPDATE_SUBMODULES_COMMAND ${UPDATE_SUBMODULES_COMMAND} --depth 1)
     ENDIF()
+    message("XXX: ${GIT_VERSION_STRING}: ${UPDATE_SUBMODULES_COMMAND}")
     IF(cmake_update_submodules MATCHES force)
       MESSAGE(STATUS "Updating submodules (forced)")
       EXECUTE_PROCESS(COMMAND ${UPDATE_SUBMODULES_COMMAND} --force
```

CI:

https://buildbot.askmonty.org/buildbot/builders/kvm-tarbake-jaunty-x86/builds/46478/steps/compile/logs/stdio

```text
XXX: 1.9.1: /usr/bin/git;submodule;update;--init;--recursive
```

Old Git: `--depth 1` wasn't appended.

Local:

```text
XXX: 2.34.1: /bin/git;submodule;update;--init;--recursive;--depth;1
```

Newer Git: `--depth 1` was appended.